### PR TITLE
updating to latest nugets

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: perf/Microsoft.Azure.WebJobs.Perf/WebJobs.Perf.csproj [hunk 1]
```diff
@@ -40,28 +40,28 @@
...
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.WindowsAzure.Configuration, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Microsoft.WindowsAzure.ConfigurationManager.3.2.3\lib\net40\Microsoft.WindowsAzure.Configuration.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.WindowsAzure.Storage, Version=7.2.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\WindowsAzure.Storage.7.2.1\lib\net40\Microsoft.WindowsAzure.Storage.dll</HintPath>
+    <Reference Include="Microsoft.WindowsAzure.Storage, Version=8.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\WindowsAzure.Storage.8.0.1\lib\net45\Microsoft.WindowsAzure.Storage.dll</HintPath>
```
Comment (brettsam) [inline perf/Microsoft.Azure.WebJobs.Perf/WebJobs.Perf.csproj]: I moved up Storage to 8.0.1 -- which is a major version upgrade. All tests passed so it looks good. It looks like there was a lot of improvements, so I'm guessing people will start wanting to use this: https://github.com/Azure/azure-storage-net/blob/master/changelog.txt

File: perf/Microsoft.Azure.WebJobs.Perf/packages.config [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+  <package id="Microsoft.Data.OData" version="5.8.2" targetFramework="net45" />
+  <package id="Microsoft.Data.Services.Client" version="5.8.2" targetFramework="net45" />
   <package id="Microsoft.WindowsAzure.ConfigurationManager" version="3.2.3" targetFramework="net45" />
   <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net45" />
-  <package id="System.Spatial" version="5.8.1" targetFramework="net45" />
-  <package id="WindowsAzure.Storage" version="7.2.1" targetFramework="net45" />
+  <package id="System.ComponentModel.EventBasedAsync" version="4.3.0" targetFramework="net45" />
+  <package id="System.Dynamic.Runtime" version="4.3.0" targetFramework="net45" />
+  <package id="System.Linq.Queryable" version="4.3.0" targetFramework="net45" />
+  <package id="System.Net.Requests" version="4.3.0" targetFramework="net45" />
```
Comment (brettsam) [inline perf/Microsoft.Azure.WebJobs.Perf/packages.config]: Microsoft.Data.Services.Client increased its dependencies in this version -- that's where these new ones came from.

File: test/Microsoft.Azure.WebJobs.Host.UnitTests/Executors/DefaultStorageCredentialsValidatorTests.cs [hunk 3]
```diff
@@ -63,7 +58,7 @@ public void StorageAccount_QueueCheckThrows_BlobOnly()
                 .Verifiable();
 
             blobClientMock.Setup(b => b.GetServicePropertiesAsync(It.IsAny<CancellationToken>()))
-                .ReturnsAsync(null);
+                .ReturnsAsync((ServiceProperties)null);
```
Comment (brettsam) [inline test/Microsoft.Azure.WebJobs.Host.UnitTests/Executors/DefaultStorageCredentialsValidatorTests.cs]: It looks like Moq added some new overloads to ReturnsAsync so it gets confused if you just return null now. Explicitly casting null to a specific object fixes it. There's a few of these sprinkled throughout this change.

File: test/Microsoft.Azure.WebJobs.ServiceBus.UnitTests/Listeners/ServiceBusListenerTests.cs [hunk 4]
```diff
@@ -86,7 +86,7 @@ public ServiceBusListenerTests()
                 {
                     await _listener.StartAsync(CancellationToken.None);
                 });
-            Assert.Equal("Unable to connect to Service Bus using HTTP connectivity mode.", ex.Message);
+            Assert.Equal("No such host is known", ex.Message);
```
Comment (brettsam) [inline test/Microsoft.Azure.WebJobs.ServiceBus.UnitTests/Listeners/ServiceBusListenerTests.cs]: It looks like Service Bus changed this error message.

File: perf/Microsoft.Azure.WebJobs.Perf/WebJobs.Perf.csproj (hunk 1 above)
Comment (xt0rted) [inline perf/Microsoft.Azure.WebJobs.Perf/WebJobs.Perf.csproj]: Even though all tests pass #985 says there's an issue with 8.0.1 but not prior versions

Comment (brettsam) [inline perf/Microsoft.Azure.WebJobs.Perf/WebJobs.Perf.csproj]: Thanks for the heads-up, I hadn't seen that before. Curious that we have no test for that -- let me take a quick look.

Comment (mathewc) [inline perf/Microsoft.Azure.WebJobs.Perf/WebJobs.Perf.csproj]: Yeah, we can't upgrade this right now. Soon though.

File: src/Microsoft.Azure.WebJobs.Logging.NuGet/WebJobs.Logging.nuspec [hunk 5]
```diff
@@ -15,7 +15,7 @@
     <tags>Microsoft Azure WebJobs Jobs Logging windowsazureofficial Web</tags>
     <dependencies>
       <dependency id="Newtonsoft.Json" version="9.0.1"/>
-      <dependency id="WindowsAzure.Storage" version="7.2.1"/>
+      <dependency id="WindowsAzure.Storage" version="8.0.1"/>
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Logging.NuGet/WebJobs.Logging.nuspec]: Undo this?

**Comment by mathewc** [comment]
Few things to fix - I made comments. Once those are all fixed and you've reviewed everything line by line to make sure all the versions are correct, just check it in :)

File: src/Microsoft.Azure.WebJobs.Logging.NuGet/WebJobs.Logging.nuspec (hunk 5 above)
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Logging.NuGet/WebJobs.Logging.nuspec]: I pushed too early -- I rolled back those as well as the BindingRedirects in the app.config files.

**Pull request closed without merging**